community/kernel/src/main/java/org/neo4j/kernel/impl/core/RelationshipProxy.java

```diff
@@ -236,6 +241,85 @@ public RelationshipType getType()
     }
 
     @Override
+    public Map<String, Object> getProperties( String... keys )
+    {
+        try ( Statement statement = actions.statement() )
+        {
+            try ( Cursor<RelationshipItem> relationship = statement.readOperations().relationshipCursor( getId() ) )
+            {
+                if ( !relationship.next() )
+                {
+                    throw new NotFoundException( "Relationship not found",
+                            new EntityNotFoundException( EntityType.RELATIONSHIP, getId() ) );
+                }
+
+                try ( Cursor<PropertyItem> propertyCursor = relationship.get().properties() )
```

Comment:
Can we create a utility-class that contains a utility method for this code that is duplicated between NodeProxy and RelationshipProxy?